zoom.sh

```diff
@@ -19,4 +19,13 @@ if [[ "$XDG_SESSION_TYPE" == "wayland" ]]; then
 	fi
 fi
 
+# Chromium's singleton mechanism is active in Zoom but doesn't actually work -
+# it uses a Qt mechanism instead. Chromium in Zoom looks for its socket in folders
+# named with the PID, so normally it won't find it. But PID namespaces mean two
+# Zoom instances can think they have the same PID, and then the second tries
+# to use this mechanism, but it does nothing (and it won't try to use the Qt
+# singleton mechanism). Deleting these sockets bypasses this.
+# https://github.com/flathub/us.zoom.Zoom/issues/445
+rm -f $HOME/.zoom/data/cefcache/*/SingletonSocket
```

Comment:
I understand the thinking here and this will work for many cases, but i don't think the hardcoded PATH here would work for all installations.

Reply:
I checked on fedora 39 and the path is referenced as: 
`.var/app/us.zoom.Zoom/.zoom/data/...`

Reply:
I'm not sure which bit of it can vary - do you see these files being created somewhere else?

Reply:
It seems that path to `.zoom` directory was changed in https://github.com/flathub/us.zoom.Zoom/pull/455 to be inside `.var` instead of directly in `home`.
After updating from flathub yesterday, I saw that my old login state was not used, and `.var/app/us.zoom.Zoom/` had a new `.zoom` directory in it. Old directory in home also stayed, but after I deleted both `~/.zoom` and `.var/app/us.zoom.Zoom` and launched Zoom again, only `.var` was used for everything including `.zoom`, and it doesn't create `~/.zoom` anymore.
Maybe your `~/.zoom` is also from old versions?

Reply:
I tested test build in this PR, making sure `~/.zoom` and `.var/app/us.zoom.Zoom` didn't exist before launching, and SSO login from Firefox works now, even though there is no `~/.zoom` in my system. Maybe path gets redirected to `~/.var/app/us.zoom.Zoom/.zoom` by flatpak automatically.
So I'm not sure whether this path for `rm` should be changed.

Reply:
> Maybe path gets redirected to ~/.var/app/us.zoom.Zoom/.zoom by flatpak automatically.

yeah, that's what `--persist` does.
The `rm` depends on that flag which is a bit more fragile than I'd like, but I not smart enough to come up with a nicer alternative.

Reply:
The code is correct. Once you remember that flatpak sandboxes apps, the behaviour is obvious.

> `xdg-{cache, config, data}` binds mount the paths from host to the per-app sandbox directory. Inside the sandbox `$XDG_CACHE_HOME, $XDG_CONFIG_HOME and $XDG_DATA_HOME` is set to `$HOME/.var/app/<app-id>/{cache, config, data}`. So this permission is not needed unless access to the host directory, bind mounted to `$HOME/.var/app/<app-id>/{cache, config, data}` is desired.
> 
> ~  https://docs.flatpak.org/en/latest/sandbox-permissions-reference.html

Reply:
> The rm depends on that flag which is a bit more fragile than I'd like, but I not smart enough to come up with a nicer alternative.

I don't think it does depend on that, because inside the sandbox, Zoom is looking at `~/.zoom`, regardless of anything that the sandbox does to redirect that. The `rm` also runs inside the sandbox, so the hardcoded path should be OK.

What would matter is if there's any config setting or environment variable that can change where Zoom looks for these files within the sandbox. I'm not aware of any, but I also haven't gone looking.